src/oscar/templates/oscar/catalogue/browse.html

```diff
@@ -64,7 +64,7 @@
         {% for value in selected_facets %}
             <input type="hidden" name="selected_facets" value="{{ value }}" />
         {% endfor %}
-        <input type="hidden" name="q" value="{{ search_form.q.value }}" />
+        <input type="hidden" name="q" value="{% firstof search_form.q.value %}" />
```

Comment:
This is a bit of a non-standard use of `firstof`. We should use the `default_if_none` filter instead.